repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRoutingServlet.scala

```diff
@@ -0,0 +1,362 @@
+    //Set the current system model, and just update the nodes.
+    val interrogator = new SystemModelInterrogator(clusterId, nodeId)
+    localCluster.set(Option(interrogator.getLocalCluster(configurationObject).orElse(null)))
+    localNode.set(Option(interrogator.getLocalNode(configurationObject).orElse(null)))
+    defaultDestination.set(Option(interrogator.getDefaultDestination(configurationObject).orElse(null)))
+    destinations.set(localCluster.get().map(_.getDestinations) match {
+      case Some(dests) => getDestinations(dests.getEndpoint.asScala.toList) ++ getDestinations(dests.getTarget.asScala.toList)
+      case None => Map.empty[String, Destination]
+    })
+    domains.set(Option(new Clusters(configurationObject)))
+
+    if (logger.underlying.isDebugEnabled) {
+      localCluster.get().foreach { cluster =>
+        val clusterId: String = cluster.getId
+        //Build a list of the nodes in this cluster, just so we know what we're doing
+        val clusterNodes = cluster.getNodes.getNode.asScala.map(node => s"${node.getId}-${node.getHostname}")
+        logger.debug("{}:{} - Nodes for this router: {}", clusterId, nodeId, clusterNodes)
+        val destStrings = destinations.get()
+          .filter { case (_, dest) => dest.isInstanceOf[DestinationEndpoint] }
+          .map { case (_, endpoint) => s"${endpoint.getId}-${endpoint.asInstanceOf[DestinationEndpoint].getHostname}" }
+        logger.debug("{}:{} - Destinations for this router: {}", clusterId, nodeId, destStrings)
+      }
+    }
+    initialized = true
+
+    def getDestinations(destList: List[Destination]): Map[String, Destination] = {
+      destList.iterator.map(dest => dest.getId -> dest).toMap
+    }
+  }
+
+  override def isInitialized: Boolean = {
+    initialized
+  }
+
+  override def service(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
+    if (!initialized) {
+      logger.error(NotInitializedMessage)
+      trySendError()
+    } else {
+      logger.trace("{} processing request...", this.getClass.getSimpleName)
+      Try(doRoute(new HttpServletRequestWrapper(req), resp)) recover {
+        case e =>
+          logger.error(FailedToRouteMessage, e)
+          trySendError()
+      }
+      logger.trace("{} returning response...", this.getClass.getSimpleName)
+    }
+
+    def trySendError(): Unit = {
+      if (!resp.isCommitted) {
+        resp.sendError(SC_INTERNAL_SERVER_ERROR, FailedToRouteMessage)
+      } else {
+        logger.error("Failed to Send Error due to response already being committed")
+      }
+    }
+  }
+
+  def doRoute(servletRequest: HttpServletRequestWrapper, servletResponse: HttpServletResponse): Unit = {
+    val attributeDestinations =
+      Option(servletRequest.getAttribute(CommonRequestAttributes.DESTINATIONS))
+        .getOrElse(List.empty[RouteDestination]).asInstanceOf[List[RouteDestination]]
+    val defaultDestinations = {
+      defaultDestination.get() match {
```

Comment:
Is it possible to not have a default destination?

Reply:
Unfortunately, yes. That is an optional field in the schema definition with the only restriction being a maximum of 1.

Reply:
Since we require at least one destination to be defined, this should probably be a v9 breaking change to require at least one default be defined.

Reply:
Story created here:
* https://repose.atlassian.net/browse/REP-7175

Reply:
Are we talking about this attribute?:
https://github.com/rackerlabs/repose/blob/master/repose-aggregator/core/repose-core-api/src/main/resources/META-INF/schema/system-model/system-model.xsd#L513-L517

That seems to read as though exactly one default is required, right?

Reply:
It counts all of the ones with the `default` attribute set to `true` and then ensures there is only one of them.
It does not ensure that there is one.

Reply:
So yes. You are right. I updated the wording in the system model and docs as well to be `one and only one`.

Reply:
Bad story deleted.